app/src/main/java/pl/droidsonroids/droidsmap/feature/office/ui/OfficeUiFeatureView.kt

```diff
@@ -58,25 +61,27 @@ class OfficeUiFeatureView(private val activity: MapActivity) : OfficeMvpView<Off
     }
 
     private fun onGroundOverlayClicked(groundOverlay: GroundOverlay) {
-        presenter.onRoomClicked()
+        presenter.onRoomClicked(Coordinates.from(groundOverlay.position))
+    }
+
+    override fun animateCameraToClickedRoom(coordinates: Coordinates) {
         val cameraPosition = CameraPosition.Builder()
                 .bearing(MAP_BEARING)
-                .target(groundOverlay.position)
+                .target(LatLng(coordinates.latitude, coordinates.longitude))
                 .zoom(MAX_MAP_ZOOM)
                 .build()
 
-        val roomImageResource = roomsList
-                .filter { it.tag == groundOverlay.tag }
-                .first()
-                .imageResource
-
         val cameraUpdate = CameraUpdateFactory.newCameraPosition(cameraPosition)
+
         with(activity.roomImage) {
             googleMap?.animateCamera(cameraUpdate, CAMERA_TRANSITION_DURATION_MILLIS, CameraListenerAdapter({
-                val imageDrawable = ContextCompat.getDrawable(activity, roomImageResource)
-                setImageDrawable(imageDrawable)
-                layoutParams.width = ((imageDrawable).intrinsicWidth * 2.2f).toInt()
-                layoutParams.height = ((imageDrawable).intrinsicHeight * 2.2f).toInt()
+                val resources = activity.resources
+                val resourceId = resources.getIdentifier("room_3", "drawable", activity.packageName)
```

Comment:
Każdy pokój ma swój drawable o unikalnej nazwie, a że nie mogę polegać na niezmienności R.id i muszę w jakiś sposób mimo to mapować instancję pokoju do konkretnego drawable, to wybieram ominięcie switcha w apce i zaciąganie tych nazw z serwera. 
Słowem: to będzie generalizowane dla wielu pokojów, a refaktor częściowy zrobiłem teraz, bo pisany test jednak też bezpośrednio dotyka problemu